python_modules/dagster/dagster/_core/definitions/declarative_scheduling/scheduling_condition.py

```diff
@@ -103,22 +106,55 @@ def __invert__(self) -> "NotAssetCondition":
         return NotAssetCondition(operand=self)
 
     @staticmethod
-    def any_deps_match(condition: "SchedulingCondition") -> "AnyDepsCondition":
+    def any_deps_match(
+        condition: "SchedulingCondition",
+        allow_keys: Optional[AbstractSet[AssetKey]] = None,
```

Comment:
Violates https://github.com/dagster-io/internal/discussions/9719

Also some future yet-to-be-written smells around "Parameter Clumping". Let's figure out a better structure here.